app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -400,4 +405,19 @@ object ComponentValidator {
     if (maybeCode === expectedCode.map(_.code)) validationSuccess else emailError
 
   }
+
+  def validateTime(fieldValue: FormComponent, time: Time)(data: FormDataRecalculated)(
+    implicit messages: Messages,
+    l: LangADT,
+    sse: SmartStringEvaluator): ValidatedType[Unit] = {
+    val timeValue = data.data.get(fieldValue.id).toSeq.flatMap(_.toSeq).filterNot(_.isEmpty).headOption
+
+    (fieldValue.mandatory, timeValue) match {
+      case (true | false, Some(vTime)) if !(Range.timeSlots(time) contains vTime) =>
+        validationFailure(fieldValue, messages("generic.error.invalid", fieldValue.label.localised.value), None)
```

Comment:
Don't use `fieldValue.label.localised.value`

Import 
```
import uk.gov.hmrc.gform.eval.smartstring.SmartStringEvaluationSyntax
```

and use `fieldValue.label.value`